src/elements/widgets/designerView/extensions/GridExtension.ts

```diff
@@ -7,32 +8,213 @@ import { OverlayLayer } from "./OverlayLayer.js";
 
 export class GridExtension extends AbstractExtension {
 
+  private _initialPoint;
+  private _initialSizes;
+  private _rects: SVGRectElement[][];
+  private _gaps: SVGRectElement[];
+  private _resizeCircles: SVGCircleElement[];
+
   constructor(extensionManager: IExtensionManager, designerView: IDesignerCanvas, extendedItem: IDesignItem) {
     super(extensionManager, designerView, extendedItem);
   }
 
   override extend() {
-    const gridInformation = CalculateGridInformation(this.extendedItem);
-    for (let gap of gridInformation.gaps) {
-      this._drawRect(gap.x, gap.y, gap.width, gap.height, 'svg-grid-gap', null, OverlayLayer.Background);
-    }
-    for (let cellRow of gridInformation.cells) {
-      for (let cell of cellRow) {
-        this._drawRect(cell.x, cell.y, cell.width, cell.height, 'svg-grid', null, OverlayLayer.Background);
+    var gridInformation = CalculateGridInformation(this.extendedItem);
+    this._rects = new Array(gridInformation.cells.length);

```

Comment:
Hab noch nie gesehen das jemand den Array constructor in javascript benutzt ;-)

Reply:
> Hab noch nie gesehen das jemand den Array constructor in javascript benutzt ;-)

Ich schon :-)